pandas/io/excel.py

```diff
@@ -68,27 +72,207 @@ def get_writer(engine_name):
         raise ValueError("No Excel writer '%s'" % engine_name)
 
 
+class BaseFile(object):
+    """Base class for excel readers
+
+    A file class can be initialized even if the engine is not installed.
+    If the engine is not installed, io_class and _open_workbook are None.
+    When attempting to use open_workbook while _open_workbook is None, the
+    relevant ImportError is raised. When calling is_ext, False is retunred when
+    the reader is not installed.
+    """
+    # added in order to deal with both Open Document Format and MS Excel
+    # fil types, see PR #9070
+
+    def __init__(self, engine=None):
+        """Set the engine name, and extension. If the engine is not installed,
+        io_class and _open_workbook are both None.
+        """
+        self.engine = engine
+        self.io_class = None
+        self._open_workbook = None
+
+    def open_workbook(self, *args, **kwargs):
+        """Explicitely load the engine again (and trigger an ImportError in the
+        process) if _open_workbook is set to None.
+        """
+        # try to load the engine again and raise import error if required
+        if self._open_workbook is None:
+            self.load_engine()
+        # just in case the user passes an already opened workbook of io_class
+        if len(args) > 0 and isinstance(args[0], self.io_class):
+            return args[0]
+        return self._open_workbook(*args, **kwargs)
+
+    def create(self, io, engine=None):
+        """Create the appropriate speadsheet book based on io and optionally
```

Comment:
this is VERY confusing what you are doing. ``.create`` is generating returning engine/book, but you aren't using the class sructure at all.

This should be a class method and simply return an instance of the appropriate sub-class, something like:

``parser = BaseFile.create(io, engine=engine)``


then it has properties for everything you need

``parser.engine, parser.book, parser.parser``, whatever.